### DISCUSSION TITLE
Runkit is dead

### DISCUSSION THREAD

**Discussion by cshaa** [community/community#141424] [npm]
The npmjs.com website has a “Try on RunKit” button next to each NPM package.

![image](https://github.com/user-attachments/assets/a8f4e3b6-1ff3-46c6-bc2b-76f65204295a)

However, trying to load up any NPM package in RunKit results in an error.

![image](https://github.com/user-attachments/assets/762e100a-1d63-48c4-9a38-1e9efc025e8d)

This has been an issue for quite some time:
 * https://discuss.runkit.com/t/lodash-module-not-found/6103
 * https://discuss.runkit.com/t/cannot-load-npm-modules/6071
 * https://discuss.runkit.com/t/new-package-not-appearing-after-a-few-days/5023
 * https://discuss.runkit.com/t/is-this-project-still-maintained/4940
 
NPM should remove the “Try on RunKit” button, and next time focus on only using open source in their supply chain, so that they can self host it in cases like these.

**Comment by danielbayley** [comment]
Yeah, maybe instead of a ‘Try on RunKit’ `<button>`, npm could instead point that button to [CodeSandbox](https://codesandbox.io), [StackBlitz](https://stackblitz.com), or an [ESM>CDN playground](https://code.esm.sh), with the module `import` filled in?

**Reply by p3k** [reply]
Please none of the three above but instead something that is similarly easy to run as Runkit was during its good days.

**Reply by kyeotic** [reply]
@p3k Do you have any examples?

**Reply by p3k** [reply]
Unfortunately I don't. It should be open source to prevent such deterioration of service.

**Reply by kyeotic** [reply]
I don't think thats something anyone here has control over. Nor is a lack of existing alternatives. I understand that runkit was very easy to use. I ended up here looking for alternatives, apparently just like you.

Thanks @danielbayley, I think Stackblitz is pretty neat. Not quite as streamlined as runkit was, but probably the best alternative that exists.

**Reply by danielbayley** [reply]
> Stackblitz is pretty neat … probably the best alternative that exists.

Actually, as mentioned [below](https://github.com/orgs/community/discussions/141424#discussioncomment-13750911), @amio put together a much better alternative! [npmjs.dev](https://npmjs.dev)

**Comment by honeymaro** [comment]
Oh my goodness, I can't even access their website anymore.

**Reply by rubengmurray** [reply]
Yeah looks like it's down completely

**Reply by rubengmurray** [reply]
https://discuss.runkit.com/t/is-this-project-still-maintained/4940

**Comment by bokub** [comment]
Runkit has been dead for months now. Can we just remove the button until a replacement is found?

**Reply by bokub** [reply]
Last commit was in 2023 FYI

https://github.com/runkitdev

**Comment by amio** [comment]
Thought about this for a long time and finally made an alternative https://npmjs.dev

**Reply by Nathan2041** [reply]
This is super useful! However, several HTML tags won't load properly. For example at [npmjs.dev/katex](https://npmjs.dev/katex) the top of the README preview is just raw HTML. After that, the rest of the preview works perfectly.

```html
<h1><a href="https://katex.org/"> <picture> <source media="(prefers-color-scheme: dark)" srcset="https://katex.org/img/katex-logo.svg"> <img alt="KaTeX" width=130 src="https://katex.org/img/katex-logo-black.svg"> </picture> </a></h1>
```

(code tags added so it didn't render on the reply)

**Reply by amio** [reply]
It's been fixed now, except the one badge from opencollective, they don't allow images on external sites.
<img width="1367" height="322" alt="image" src="https://github.com/user-attachments/assets/7e5c8b44-04c9-4892-9ec0-dbd26d610e3d" />

**Comment by tofrankie** [comment]
😔

**Comment by saz419** [comment]
Are there any open-source alternatives similar to Runkit that we can host ourselves?

**Reply by marek-saji** [reply]
npmjs.dev from @amio is: https://github.com/amio/npmjs.dev

**Comment by NeverBehave** [comment]
While people mentioning about npmjs.dev, one of the missing feature is runkit allows you to expose a lightweight web server with logs. 

I moved to [cloudflare workers](https://workers.cloudflare.com/) for this use case.

**Comment by tutyamxx** [comment]
Sorry to necro the thread I made one also https://github.com/tutyamxx/NpmRunner (still wip, i'll add more support for different things later). Now has a domain https://npmrunner.com

**Comment by idleberg** [comment]
npmjs.com could be in a much better shape if they'd open-source it. It feels like there hasn't been a single change in years and things start to smell.
